utils/bump_version.py

```diff
@@ -0,0 +1,76 @@
+import subprocess
+from pathlib import Path
+
+import semver
+import tomlkit
+
+from utils.check_pypi import get_releases
+
+
+def get_pyproject_version():
+    root_dir = Path(__file__, "..").resolve().parent
+    with open(root_dir / "pyproject.toml", "rb") as fh:
+        project = tomlkit.load(fh)
+        return project["tool"]["poetry"]["version"]
+
+
+def _get_next_version(version, versions, bump_version):
+    sorted_versions = sorted([semver.Version.parse(v) for v in versions])
+    current_version = sorted_versions[-1]
+    if semver.Version.parse(version) > current_version:
+        return version
+    if bump_version:
+        return str(current_version.bump_patch())
+    return str(current_version)
+
+
+def bump_version_if_required(versions, bump_version):
+    pyproject_version = get_pyproject_version()
+    correct_version = _get_next_version(pyproject_version, versions, bump_version)
+
+    if correct_version == pyproject_version:
+        return 0
+
+    root_dir = Path(__file__, "..").resolve().parent
+    with open(root_dir / "pyproject.toml", "rb") as fh:
+        project = tomlkit.load(fh)
+
+    print(f"{'Bumping' if bump_version else 'Setting'} the project version to {correct_version}")
+    project["tool"]["poetry"]["version"] = correct_version
+
+    with open(root_dir / "pyproject.toml", "w") as fh:
+        tomlkit.dump(project, fh)
+
+    return 1
+
+
+def version_should_be_bumped(files):
+    for f in files:
+        filepath = Path(f)
+        if filepath.name == "README.md":
+            continue
+        parents = list(str(p) for p in filepath.parents)
+        # Each parent is the full path from the root.
+        # We don't need to bump if it is just tests that have changed:
+        if "tests" in parents:
+            continue
+
+        return True
+
+    return False
+
+
+def get_changed_files():
```

Comment:
```suggestion
_def get_changed_files():
```

Would need to change where it's used too obviously.

Reply:
I put the `_` in the wrong place.